repository.py

```diff
@@ -6,16 +6,22 @@ def all(resource):
         
 def retrieve(resource, id):
     """For GET requests to a single resource"""
-    return method_mapper[resource]["single"](id)
+    if isinstance(id, dict):
```

Comment:
Checking types is non-Pythonic. It is recommended that you use `try..catch` blocks instead.

```py
try:
    parameter = list(id.keys())[0]
    value = list(id.values())[0][0]
    return  method_mapper[resource][parameter](value)
except IndexError:
    return method_mapper[resource]["single"](id)
```